jwks/key.go

```diff
@@ -0,0 +1,145 @@
+package jwks
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"encoding/base64"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math/big"
+)
+
+const typeRSA = "RSA"
+
+var supportedKeyTypes = map[string]struct{}{
+	typeRSA: {},
+}
+
+// JWK defines JSON Web Key structure.
+type JWK struct {
```

Comment:
I suggest adding unit tests for `DecodePublicKey()` and `DecodePrivateKey()` methods to ensure their correctness and reliability.